### PR TITLE
refactor(server)!: rename events table to workflow_log with filtered persistence

### LINKED ISSUE #393: Refactor events table to workflow_log (slim persistence)
## Summary

Replace the `events` table with a slim `workflow_log` table that only persists high-level workflow events. Verbose trace events become stream-only (in-memory).

**Breaking change:** Requires deleting `~/.amelia/` (clean break).

**Prerequisite for:** PostgreSQL migration

## Motivation

The current `events` table stores everything—thinking blocks, tool calls, streaming chunks, agent output—resulting in thousands of rows per workflow. Most of this data is only useful during live execution for the dashboard's real-time updates.

| Metric | Before | After |
|--------|--------|-------|
| Rows per workflow | Thousands | 10-50 |
| Columns | 16 | 9 |
| Retention settings | 2 (`log` + `trace`) | 1 (`log` only) |

## Design

See full design document: `docs/plans/2026-01-31-events-to-workflow-log.md`

### Persisted Event Types (workflow_log)

- **Lifecycle**: `workflow_created`, `workflow_started`, `workflow_completed`, `workflow_failed`, `workflow_cancelled`
- **Stages**: `stage_started`, `stage_completed`
- **Approval**: `approval_required`, `approval_granted`, `approval_rejected`
- **Artifacts**: `file_created`, `file_modified`, `file_deleted`
- **Review**: `review_requested`, `review_completed`, `revision_requested`
- **Tasks**: `task_started`, `task_completed`, `task_failed`
- **System**: `system_error`, `system_warning`
- **Oracle**: `oracle_consultation_started`, `oracle_consultation_completed`, `oracle_consultation_failed`
- **Brainstorm**: `brainstorm_session_created`, `brainstorm_session_completed`, `brainstorm_artifact_created`

### Stream-only (not persisted)

- Claude trace events (`claude_thinking`, `claude_tool_call`, etc.)
- Oracle trace events
- Brainstorm trace events
- Streaming events (`stream`, `agent_message`)

## Files to Change

| File | Change |
|------|--------|
| `amelia/server/database/connection.py` | Replace `events` DDL with `workflow_log` |
| `amelia/server/database/repository.py` | Add `PERSISTED_TYPES`, filter in `save_event()`, rename table references |
| `amelia/server/models/events.py` | Export `PERSISTED_TYPES` frozenset |
| `amelia/server/lifecycle/retention.py` | Remove trace retention logic |
| `amelia/server/config.py` | Remove `trace_retention_days` setting |
| `tests/integration/test_workflow_log.py` | New E2E test for event persistence |
| `tests/unit/test_event_filtering.py` | Unit tests for edge cases |

### Primary: End-to-end integration test

One comprehensive test that runs a real workflow with only the LLM HTTP boundary mocked, verifying:
- Lifecycle events are persisted
- Trace events are NOT persisted
- Row count is reasonable (~10-50, not thousands)

### Secondary: Unit tests for edge cases

- `test_save_event_filters_stream_only_types`
- `test_workflow_log_schema_constraints`

## Configuration Changes

| Setting | Before | After |
|---------|--------|-------|
| `log_retention_days` | Keep (default 30) | Keep (default 30) |
| `trace_retention_days` | Remove | N/A |
| `stream_tool_results` | Keep | Keep (controls WebSocket streaming) |

### PR DESCRIPTION
**Pull request by anderskev**
## Summary

Replace the catch-all `events` table with a focused `workflow_log` table that only persists meaningful workflow events. Trace and streaming events are now broadcast-only via WebSocket, eliminating high-volume low-value database writes.

### Changed
- Rename `events` table to `workflow_log` with a simplified schema (dropped `correlation_id`, `tool_name`, `tool_input_json`, `trace_id`, `parent_id` columns)
- Introduce `PERSISTED_TYPES` set for explicit persistence control — repository silently skips non-persisted event types
- Replace `TRACE` level with `ERROR`/`WARNING` for a cleaner 4-level system (`DEBUG`, `INFO`, `WARNING`, `ERROR`)
- Simplify `EventBus` by removing trace-level filtering logic and `configure()` method
- Catch specific `sqlite3.Error` instead of bare `Exception` in database code

### Removed
- `trace_retention_days` setting from server config, CLI, API, and dashboard UI
- Trace retention cleanup logic from `LogRetentionService`
- `EventLevel.TRACE` enum value
- Trace-specific columns from the database schema

## Motivation

The events table was storing every event including high-volume trace and streaming events, leading to unnecessary database bloat. Most of these events are only useful in real-time (via WebSocket) and have no value once the workflow completes. By filtering at the persistence boundary, we keep the database lean while preserving real-time visibility.

Closes #393

## Testing

- [x] Unit tests updated across 13 test files
- [x] New `tests/unit/test_event_filtering.py` added for persistence filtering
- [x] Schema tests updated for `workflow_log` table
- [x] Dashboard tests updated to remove trace retention references
- [x] All existing tests pass

## Breaking Changes

- **Database schema**: `events` table renamed to `workflow_log` with fewer columns. Existing databases will need migration (handled automatically by schema init).
- **Server config**: `trace_retention_days` setting removed. Any existing configuration referencing this setting will be ignored.
- **API**: `trace_retention_days` field removed from server settings endpoints.

---

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: amelia/server/database/connection.py [hunk 1]
```diff
@@ -277,22 +278,17 @@ async def ensure_schema(self) -> None:
...
-            CREATE TABLE IF NOT EXISTS events (
+            CREATE TABLE IF NOT EXISTS workflow_log (
                 id TEXT PRIMARY KEY,
                 workflow_id TEXT NOT NULL REFERENCES workflows(id) ON DELETE CASCADE,
                 sequence INTEGER NOT NULL,
                 timestamp TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
-                agent TEXT NOT NULL,
                 event_type TEXT NOT NULL,
-                level TEXT NOT NULL DEFAULT 'debug',
+                level TEXT NOT NULL CHECK (level IN ('info', 'warning', 'error', 'debug')),
```
[the remark this answers is not in the document]
Comment (anderskev) [reply amelia/server/database/connection.py]: @coderabbitai Fixed. Updated `EventLevel` type to `'info' | 'warning' | 'error' | 'debug'` and propagated the change across all dashboard references:

- `dashboard/src/types/index.ts` — type definition and docstring
- `dashboard/src/types/__tests__/index.test.ts` — level values and count
- `dashboard/src/mocks/infinite-mode.ts` — removed stale `traceEvents` set
- `dashboard/src/pages/LogsPage.tsx` — filter changed from `'trace'` to `'debug'`
- `dashboard/src/pages/__tests__/LogsPage.test.tsx` — test helper and assertions
- `dashboard/src/components/activity/useActivityLogGroups.ts` — removed dead `!= 'trace'` filter
- `dashboard/src/components/activity/__tests__/useActivityLogGroups.test.ts` — updated test
- `dashboard/src/components/__tests__/ActivityLog.test.tsx` — updated test

All 822 dashboard tests pass, TypeScript compiles clean, production build succeeds.
